client/keystore/src/local.rs

```diff
@@ -234,6 +236,72 @@ impl Keystore for LocalKeystore {
 		Ok(sig)
 	}
 
+	#[cfg(feature = "bandersnatch-experimental")]
+	fn bandersnatch_public_keys(&self, key_type: KeyTypeId) -> Vec<bandersnatch::Public> {
+		self.public_keys::<bandersnatch::Pair>(key_type)
+	}
+
+	/// Generate a new pair compatible with the 'bandersnatch' signature scheme.
+	///
+	/// If `[seed]` is `Some` then the key will be ephemeral and stored in memory.
+	#[cfg(feature = "bandersnatch-experimental")]
+	fn bandersnatch_generate_new(
+		&self,
+		key_type: KeyTypeId,
+		seed: Option<&str>,
+	) -> std::result::Result<bandersnatch::Public, TraitError> {
+		self.generate_new::<bandersnatch::Pair>(key_type, seed)
+	}
+
+	#[cfg(feature = "bandersnatch-experimental")]
+	fn bandersnatch_sign(
+		&self,
+		key_type: KeyTypeId,
+		public: &bandersnatch::Public,
+		msg: &[u8],
+	) -> std::result::Result<Option<bandersnatch::Signature>, TraitError> {
```

Comment:
What's the point in Result<Option>? What would Ok(None) mean?

Reply:
It'll mean the key does not exist according to https://github.com/paritytech/substrate/pull/14412/files/13ad9ad4f7ac43a01d3ac4245ab9f44c471ceb91#diff-50b4101719046fd4020e8653e988eaf5ffbf27583dccb3bc9acadef9f5236c23L57

I've no idea if another signing error can yield `Ok(None)` too, but likely they want errors to be serialized acording to https://github.com/paritytech/substrate/blob/master/primitives/keystore/src/lib.rs#L33